penguin/tools/image_tools.py

```
@@ -0,0 +1,73 @@
+    def execute(
+        self,
+        path: str,
+        prompt: Optional[str] = None,
+        max_dim: Optional[int] = None,
+    ) -> Dict[str, Any]:
+        if not path:
+            return {"error": "path is required"}
+
+        try:
+            image_path = enforce_allowed_path(Path(path), root_pref="auto")
+        except Exception as exc:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)